go/vt/wrangler/schema.go

```diff
@@ -273,6 +277,60 @@ func (wr *Wrangler) ValidateSchemaKeyspace(ctx context.Context, keyspace string,
 	return nil
 }
 
+// ValidateVSchema compares the schema of each primary tablet in "keyspace/shards..." to the vschema and errs if there are differences
+func (wr *Wrangler) ValidateVSchema(ctx context.Context, keyspace string, shards []string, excludeTables []string, includeViews bool) error {
+	vschm, err := wr.ts.GetVSchema(ctx, keyspace)
+	if err != nil {
+		return fmt.Errorf("GetVSchema(%s) failed: %v", keyspace, err)
+	}
+
+	shardFailures := concurrency.AllErrorRecorder{}
+	var wg sync.WaitGroup
+	wg.Add(len(shards))
+
+	for _, shard := range shards {
+		go func(shard string, shardFailures *concurrency.AllErrorRecorder) {
```

Comment:
You shouldn't need to pass a reference to the recorder here; it's in scope in the closure of the goroutine (similar to how you're using the wait group).

Reply:
I actually got linting errors when I did not pass in the recorder or shard which is why I passed them in

```
Linting go/vt/wrangler
go/vt/wrangler/schema.go:289:45: loopclosure: loop variable shard captured by func literal (govet)
			si, err := wr.ts.GetShard(ctx, keyspace, shard)
			                                         ^
go/vt/wrangler/schema.go:291:83: loopclosure: loop variable shard captured by func literal (govet)
				shardFailures.RecordError(fmt.Errorf("GetShard(%v, %v) failed: %v", keyspace, shard, err))
				                                                                              ^
go/vt/wrangler/schema.go:297:45: loopclosure: loop variable shard captured by func literal (govet)
					excludeTables, includeViews, keyspace, shard, err,
					                                       ^
```